docs/project-docs/developer-guide.md

```diff
@@ -66,3 +66,8 @@ One can build in Debug or Release mode from the root by doing `build.cmd -Releas
 
 We only support 64-bit binaries right now.
 
+### Updating manifest and ep-list files
+
+During development, there may arise a need to update the current baseline `core_manifest.json` and `core_ep-list.tsv` files.
+
+This is done through running `RegenerateEntryPointCatalog` in `machinelearning/test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs`. This can be done by unskipping `RegenerateEntryPointCatalog`, running `RegenerateEntryPointCatalog`, and then pushing the changes to these files.
```

Comment:
May want more details to remove room for misunderstandings:
```suggestion
Steps:
1. Unskip the `RegenerateEntryPointCatalog` unit test in `test/Microsoft.ML.Core.Tests/UnitTests/TestEntryPoints.cs`. This can be done by temporarily commenting out the skip attribute on the unit test for `RegenerateEntryPointCatalog` (`[Fact(Skip = "Execute this test if you want to regenerate the core_manifest and core_ep_list files")]`)
2. Run the unit tests (`build.cmd -runTests`)
3. Verify the changes to `core_manifest.json` and `core_ep-list.tsv` are correct
4. Re-enable the skip attribute on the `RegenerateEntryPointCatalog` test
5. Commit the updated `core_manifest.json` and `core_ep-list.tsv` files to your branch
```

Reply:
@ganik : Is the above the recommended method of updating the `core_manifest.json` and `core_ep-list.tsv` files?

Reply:
I think so. This is how I have always done it.

Step 2 - Running all tests - might be unnecessary in cases of a simple rename, only running `RegenerateEntryPointCatalog` would suffice. In other situations, we might want to run all tests, for example if there was also a change in the entrypoint infrastructure.

Should we call this out or is it better to err on the safe side and always run all tests?

Reply:
yes

---
In reply to: [372553266](https://github.com/dotnet/machinelearning/pull/4727#discussion_r372553266) [](ancestors = 372553266)